clientapi/routing/account_data.go

```diff
@@ -79,11 +80,26 @@ func SaveAccountData(
 
 	defer req.Body.Close() // nolint: errcheck
 
+	if req.Body == http.NoBody {
+		return util.JSONResponse{
+			Code: http.StatusBadRequest,
+			JSON: jsonerror.NotJSON("Content not JSON"),
+		}
+	}
+
 	body, err := ioutil.ReadAll(req.Body)
 	if err != nil {
 		return httputil.LogThenError(req, err)
 	}
 
+	var rawJson json.RawMessage
+	if err = json.Unmarshal(body, &rawJson); err != nil {
```

Comment:
Looks like [`json.Valid()`](https://golang.org/pkg/encoding/json/#Valid) should be enough.

Reply:
Done